Vagrant: deprecate NETNEXT=*, use KERNEL=netnext to use net-next kernel for CI & dev VMs

qmonnet opened the pull request:
There are currently three different ways to use variables to tell ginkgo to use the net-next kernel for end-to-end tests: `NETNEXT=true`, `NETNEXT=1`, and `KERNEL=net-next`. This is getting confusing, and prone to omissions on updates.

Let's deprecate the `NETNEXT` environment variable. `KERNEL=netnext` (__[edit]__ Note the simplification from `net-next` to `netnext`) is better aligned with the method for selecting the 4.19 kernel as well.

__[edit]__ Let's align the setup for the development VMs as well, and use `KERNEL=netnext` instead of the `NETNEXT` variants.

qmonnet committed:
test: deprecate NETNEXT=, use KERNEL=netnext to use net-next kernel

There are currently three different ways to use variables to tell ginkgo
to use the net-next kernel for end-to-end tests: NETNEXT=true,
NETNEXT=1, and KERNEL=netnext. This is getting confusing, and prone to
omissions on updates.

Let's deprecate the NETNEXT environment variable. "KERNEL=netnext" is
better aligned with the method for selecting the 4.19 kernel as well.

Note that the development setup still relies on a NETNEXT variable to
select the net-next kernel. This is not changed by the current commit.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

pchaigno commented on test/helpers/utils.go:
```diff
@@ -496,7 +496,11 @@ func failIfContainsBadLogMsg(logs, label string, blacklist map[string][]string)
+		return true
+	}
+	netNext = os.Getenv("KERNEL")
+	return netNext == "net-next"
```
Could this be just `netnext`? :pray:

qmonnet replied:
No objection, I'll update the PR and then we'll see what folks say. Let's keep this in one PR, and avoid changing the commands twice in a row for people who are still using `NETNEXT=`.

pchaigno reviewed (approved):
LGTM :tada: 

I would just really prefer `netnext` to `net-next`, but I can send a PR for that afterward if everyone agrees.

aditighag commented on test/Vagrantfile:
```diff
@@ -40,7 +40,7 @@ $CILIUM_REGISTRY = ENV['CILIUM_REGISTRY'] || ""
 $MEMORY = (ENV['VM_MEMORY'] || "4096").to_i
 $CPU = (ENV['VM_CPUS'] || "2").to_i
 
-if ENV['NETNEXT'] == "true" || ENV['NETNEXT'] == "1" || ENV['KERNEL'] == "net-next" then
```
It would be nice if the environment variables for the dev VM and test VM are same. I personally would prefer to continue to use `NETNEXT` for the dev VM (I already have it stored in my IDE settings/scripts), and hence, test VM as well.

pchaigno replied:
I think the goal is to try and use a single env. variable for everything. It's a bit confusing to see both `NETNEXT` and `KERNEL`. Clearly it needs to be `KERNEL` (imagine `NETNEXT=419` :laughing:).

I'm happy if we can move to just `KERNEL` for both the test and the development VM. Not sure if we can do that without first deprecating `NETNEXT` for the development VM; maybe a message in #development would be enough :thinking:

qmonnet replied:
So I've gone the most “extreme” path, and updated the variables both for CI and development setups. The new version uses `KERNEL=netnext` everywhere, `NETNEXT` variants are all removed (Sorry Aditi, I agree with Paul on `KERNEL` being more consistent). I'll link the PR on Slack so we can gather more feedback.

qmonnet committed:
test: rename KERNEL option "net-next" to "netnext"

Make it easier to type. No dash. "netnext", that's all there is to it.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
Vagrant: deprecate NETNEXT=1|true in favour of KERNEL=netnext

Let's harmonise the way we select the net-next kernel between the CI
environment and the development setup. Let's go with "KERNEL=", which
sets the version (possibly other than net-next) for the CI setup.

Mark NETNEXT as unsupported and error out if the user attempts to pass
it, to avoid starting VMs with the wrong kernel.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

pchaigno commented on test/k8sT/manifests/externalIPs/README.md:
````diff
@@ -10,7 +10,7 @@ will run with node-port feature enabled and without kube-proxy.
 
 ```bash
-NETNEXT=1 NFS=1 NWORKERS=1 K8S=1 ./contrib/vagrant/start.sh
+KERNEL=net-next NFS=1 NWORKERS=1 K8S=1 ./contrib/vagrant/start.sh
````
```suggestion
KERNEL=netnext NFS=1 NWORKERS=1 K8S=1 ./contrib/vagrant/start.sh
```

qmonnet replied:
Thanks, the changes were lost when I reordered commits :(.

pchaigno commented on Documentation/contributing/testing/e2e.rst:
```diff
@@ -148,7 +148,7 @@ supported version of Kubernetes, run the test suite with the following format:
 .. note::
 
-   When provisioning VMs with the net-next kernel (``NETNEXT=1``) on
+   When provisioning VMs with the net-next kernel (``KERNEL=net-next``) on
```
```suggestion
   When provisioning VMs with the net-next kernel (``KERNEL=netnext``) on
```

pchaigno replied:
Commit message also needs to be updated.

pchaigno reviewed (commented):
There are a couple of places where it still uses `net-next`.

qmonnet commented on jenkinsfiles/ginkgo-runtime-kernel.Jenkinsfile:
```diff
@@ -14,9 +14,9 @@ pipeline {
+        KERNEL="""${sh(
             returnStdout: true,
-            script: 'if [ "${JobKernelVersion}" = "net-next" ]; then echo -n "1"; else echo -n "0"; fi'
+            script: 'if [ "${JobKernelVersion}" = "net-next" ]; then echo -n "netnext"; else echo -n ""; fi'
```
I don't know if this should be updated as well? I'm not sure where this variable is set and used.
```suggestion
            script: 'if [ "${JobKernelVersion}" = "netnext" ]; then echo -n "netnext"; else echo -n ""; fi'
```
